src/sql/src/rbac.rs

```diff
@@ -38,40 +38,45 @@ use crate::session::user::{MZ_SUPPORT_ROLE_ID, MZ_SYSTEM_ROLE_ID, SUPPORT_USER,
 use crate::session::vars::SystemVars;
 
 /// Common checks that need to be performed before we can start checking a role's privileges.
-fn rbac_preamble(
+macro_rules! rbac_preamble {
+    ($catalog:expr, $session_meta:expr) => {
+        // PostgreSQL allows users that have their role dropped to perform some actions,
+        // such as `SET ROLE` and certain `SELECT` queries. We haven't implemented
+        // `SET ROLE` and feel it's safer to force to user to re-authenticate if their
+        // role is dropped.
+        if $catalog
+            .try_get_role(&$session_meta.role_metadata().current_role)
+            .is_none()
+        {
+            return Err(UnauthorizedError::ConcurrentRoleDrop(
+                $session_meta.role_metadata().current_role.clone(),
+            ));
+        };
+        if $catalog
+            .try_get_role(&$session_meta.role_metadata().session_role)
+            .is_none()
+        {
+            return Err(UnauthorizedError::ConcurrentRoleDrop(
+                $session_meta.role_metadata().session_role.clone(),
+            ));
+        };
+        if $catalog
+            .try_get_role(&$session_meta.role_metadata().authenticated_role)
+            .is_none()
+        {
+            return Err(UnauthorizedError::ConcurrentRoleDrop(
+                $session_meta.role_metadata().authenticated_role.clone(),
+            ));
+        };
+    };
+}
```

Comment:
Just realized that there's no need for this to be a macro, I'll switch to a method.